### PR TITLE
Added PlaceholderText functionality 

### PR DESCRIPTION
**Pull request by stefanov-stefan**
Recreating https://github.com/dotnet/winforms/pull/96

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/TextBox.cs [hunk 1]
```diff
@@ -843,9 +844,86 @@ internal void SetAutoComplete(bool reset)
...
-	//-------------------------------------------------------------------------------------------------
         
+        /// <summary>
+        ///  Gets or sets the text that is displayed when the control has no Text and is not on focus.
+        /// </summary>
+        [
+        Localizable(true),
+        DefaultValue(""),
+        SRDescription(nameof(SR.TextBoxPlaceholderTextDescr)),
+        Browsable(true), EditorBrowsable(EditorBrowsableState.Always)
```
Comment (jnm2) [inline src/System.Windows.Forms/src/System/Windows/Forms/TextBox.cs]: All properties are `[Browsable(true), EditorBrowsable(EditorBrowsableState.Always)]` by default.

File: src/System.Windows.Forms/src/System/Windows/Forms/TextBox.cs [hunk 2]
```diff
@@ -843,9 +844,86 @@ internal void SetAutoComplete(bool reset)
...
         }
-	
-	//-------------------------------------------------------------------------------------------------
         
+        /// <summary>
+        ///  Gets or sets the text that is displayed when the control has no Text and is not on focus.
+        /// </summary>
+        [
+        Localizable(true),
+        DefaultValue(""),
```
Comment (jnm2) [inline src/System.Windows.Forms/src/System/Windows/Forms/TextBox.cs]: Wouldn't it be `DefaultValue(null)`?

File: src/System.Windows.Forms/src/System/Windows/Forms/TextBox.cs (hunk 1 above)
Comment (stefanov-stefan) [reply src/System.Windows.Forms/src/System/Windows/Forms/TextBox.cs]: Added these for visibility purposes, however, you are right and will remove them.

File: src/System.Windows.Forms/src/System/Windows/Forms/TextBox.cs (hunk 2 above)
Comment (stefanov-stefan) [reply src/System.Windows.Forms/src/System/Windows/Forms/TextBox.cs]: Yep.

**Comment by JuditRose** [comment]
LGTM, Do we need any internal manual testing or functional test case to add?

**Comment by zsd4yr** [comment]
LGTM; @KlausLoeffelmann @AdamYoblick Is there anything else either of you would like to see here?

**Comment by AdamYoblick** [comment]
If it got API approved, fine with me. :)

**Comment by karelz** [comment]
We should use "Fixes #..." format in top post (I just added it) + have some description there ;)

**Pull request merged**